### PR TITLE
chore: webui disable "continue trial" when there are no checkpoints [DET-3850]

### PR DESCRIPTION
**Pull request by apizzini**
## Description

This PR addresses the request of the JIRA issues. 

## Test Plan

App working fine.

## Commentary (optional)

Schermata 2020-09-15 alle 11 09 01

## Checklist

- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](../docs/release-notes/README.md) for details.

### CODE REVIEW

File: webui/react/src/pages/TrialDetails/TrialActions.tsx [hunk 1]
```diff
@@ -15,6 +15,7 @@ export enum Action {
 }
 
 interface Props {
+  experiment: ExperimentDetails,
```
Comment (hkang1) [inline webui/react/src/pages/TrialDetails/TrialActions.tsx:18]: **suggestion:** looks like we only really need the trials. Maybe just this?
```
  trials: TrialItem[],
```

File: webui/react/src/pages/TrialDetails/TrialActions.tsx [hunk 2]
```diff
@@ -43,11 +44,20 @@ const TrialActions: React.FC<Props> = ({ trial, onClick, onSettled }: Props) =>
     setButtonStates(state => ({ ...state, tensorboard: false }));
   }, [ trial.id, onSettled ]);
 
+  let trialCompletedCheckpointSum = 0;
+  experiment.trials.map((trial) => {
+    trialCompletedCheckpointSum += trial.numCompletedCheckpoints;
+  });
```
Comment (hkang1) [inline webui/react/src/pages/TrialDetails/TrialActions.tsx:50]: **suggestion:** This is a good candidate for `useMemo` where you can memoize the result of an "expensive" operation.
```
const trialCompletedCheckpointSum = useMemo(() => {
  return trials.reduce((acc, trial) => acc + trial.numCompletedCheckpoints, 0);
}, [ trials ]);
```

File: webui/react/src/pages/TrialDetails/TrialActions.tsx [hunk 3]
```diff
@@ -43,11 +44,20 @@ const TrialActions: React.FC<Props> = ({ trial, onClick, onSettled }: Props) =>
...
   const actionButtons: ConditionalButton<TrialDetails>[] = [
     {
-      button: <Button
-        key={Action.Continue}
-        onClick={onClick(Action.Continue)}>Continue Trial</Button>,
+      button: (trialCompletedCheckpointSum > 0 ? (
+        <Button key={Action.Continue} onClick={onClick(Action.Continue)}>Continue Trial</Button>
+      ) : (
+        <Tooltip key={Action.Continue} title={'No checkpoints found. Cannot continue trial.'}>
+          <Button disabled={true}>Continue Trial</Button>
```
Comment (hkang1) [inline webui/react/src/pages/TrialDetails/TrialActions.tsx:58]: **nit:** I think you can just have `disabled` without the explicit `={true}` defined, but doesn't hurt to be explicit.

**Review by hkang1 — COMMENTED** [review]
Great job with the checkpoint tallying! Some minor suggestions around memoization and what exact props to pass in, but overall looks good!

**Review by hkang1 — APPROVED** [review]
Approving this since the changes are really non-blocking and customers can really use this (target release of tomorrow morning)

**Comment by apizzini** [comment]
> Approving this since the changes are really non-blocking and customers can really use this (target release of tomorrow morning)

Thanks for the feedbacks. Should be good now.

**Comment by hkang1** [comment]
> > Approving this since the changes are really non-blocking and customers can really use this (target release of tomorrow morning)
> 
> Thanks for the feedbacks. Should be good (even better) now.

Looks great @apizzini ! 👍

**Pull request merged**